Source/ACE/Entity/Container.cs

```diff
@@ -26,48 +29,111 @@ public Container(ObjectType type, ObjectGuid guid, string name, ushort weenieCla
+            ValidLocations = (EquipMask?)baseAceObject.ValidLocations;
+            Value = baseAceObject.Value;
+            Workmanship = baseAceObject.Workmanship;
+
+            PhysicsData.SetPhysicsDescriptionFlag(this);
+            WeenieFlags = SetWeenieHeaderFlag();
+            WeenieFlags2 = SetWeenieHeaderFlag2();
+
+            baseAceObject.AnimationOverrides.ForEach(ao => ModelData.AddModel(ao.Index, ao.AnimationId));
+            baseAceObject.TextureOverrides.ForEach(to => ModelData.AddTexture(to.Index, to.OldId, to.NewId));
+            baseAceObject.PaletteOverrides.ForEach(po => ModelData.AddPalette(po.SubPaletteId, po.Offset, po.Length));
+            ModelData.PaletteGuid = baseAceObject.PaletteId;
+        }
+
         // Inventory Management Functions
         public virtual void AddToInventory(WorldObject inventoryItem)
         {
             if (!inventory.ContainsKey(inventoryItem.Guid))
             {
                 inventory.Add(inventoryItem.Guid, inventoryItem);
             }
-
-            Burden += inventoryItem.Burden;
-            inventoryItem.ContainerId = Guid.Full;
-            if (inventoryItem.Location != null)
-                LandblockManager.RemoveObject(inventoryItem);
-            inventoryItem.PhysicsData.PhysicsDescriptionFlag &= ~PhysicsDescriptionFlag.Position;
-            inventoryItem.PositionFlag = UpdatePositionFlag.None;
-            inventoryItem.PhysicsData.Position = null;
-            inventoryItem.Location = null;
-            inventoryItem.WeenieFlags = inventoryItem.SetWeenieHeaderFlag();
         }
 
         public virtual void RemoveFromInventory(ObjectGuid inventoryItemGuid)
         {
-            WorldObject inventoryItem = GetInventoryItem(inventoryItemGuid);
-            if (Burden >= inventoryItem.Burden)
-                Burden -= inventoryItem.Burden;
-            else
-                Burden = 0;
-
-            // FIXME(ddevec): RemoveFromInventory should only be responsible for removing the item from the inventory, not placing it on the world!
-            inventoryItem.Location = Location.InFrontOf(1.0f);
-            // TODO: Write a method to set this based on data.
-            inventoryItem.PositionFlag = UpdatePositionFlag.Contact
-                                         | UpdatePositionFlag.Placement
-                                         | UpdatePositionFlag.ZeroQy
-                                         | UpdatePositionFlag.ZeroQx;
-
-            inventoryItem.PhysicsData.PhysicsDescriptionFlag = inventoryItem.PhysicsData.SetPhysicsDescriptionFlag(inventoryItem);
-            inventoryItem.ContainerId = null;
-            inventoryItem.Wielder = null;
-            inventoryItem.WeenieFlags = inventoryItem.SetWeenieHeaderFlag();
             if (inventory.ContainsKey(inventoryItemGuid))
             {
                 inventory.Remove(inventoryItemGuid);
+                Burden = UpdateBurden();
+            }
+            else
```

Comment:
I've seen enough of this inventory-ownership-check to think that we should have a method for it.  Something along the lines of "HasItem(uint itemGuid)".  Especially now that it's 18 lines of code instead of 2.